### ISSUE TITLE
Validation of incoming requests is absent

### ISSUE THREAD

**Issue by jaredbischof** [goadesign/goa#608]
goagen does not appear to generate a Validate() function in app/user_types.go from my design/resources.go file.  I've attached both files in txt format.  The command I'm running to generate the code is:

`goagen app -d github.com/<mypath>/design`

[resources.go.txt](https://github.com/goadesign/goa/files/354877/resources.go.txt)
[user_types.go.txt](https://github.com/goadesign/goa/files/354881/user_types.go.txt)

**Comment by verdverm** [comment]
look in the generated app/context.go file, validation happens there.

**Comment by jaredbischof** [comment]
There should be Validate() functions in both the app/context.go and app/user_types.go generated files.

**Comment by jaredbischof** [comment]
Sorry @verdverm, you were correct with your response. The broader issue I was trying to understand was why, when my Payload definition contained a required field, was I still able to do a "create" without including that required field. What I failed to realize was that by using the auto-generated client to do the create, even though I did not specify the required field, the client code was automatically adding the required field with an empty string to my POST request.
